### PR TITLE
{monitor}`az monitor action-group`: Migrate from codegen V2

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

az monitor action-group

**Description**

**Testing Guide**

**History Notes**

[monitor] `az monitor action-group`: Migrate from codegen V2

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
monitor

File: src/azure-cli-core/azure/cli/core/aaz/_arg_action.py [hunk 1]
```diff
@@ -483,6 +483,112 @@ def format_data(cls, data):
         raise AAZInvalidValueError(f"list type value expected, got '{data}'({type(data)})")
 
 
+class AAZActionGroupReceiverArgAction(AAZCompoundTypeArgAction):
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/aaz/_arg_action.py:486]: This is monitor special arg arction, please move this code in monitor module.

File: src/azure-cli-core/azure/cli/core/aaz/_arg.py [hunk 2]
```diff
@@ -449,6 +449,37 @@ def _type_in_help(self):
...
+
+    def _build_cmd_action(self):
+        class Action(AAZActionGroupReceiverArgAction):
+            _schema = self  # bind action class with current schema
+
+        return Action
+
+    @property
+    def _type_in_help(self):
+        return f"List<{self.Element._type_in_help}>"
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/aaz/_arg.py:480]: Please move it in monitor module

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_action_groups.py [hunk 3]
```diff
@@ -130,7 +133,7 @@ def test_monitor_action_group_location(self, resource_group):
             JMESPathCheck('length(smsReceivers)', 0),
             JMESPathCheck('length(webhookReceivers)', 0),
             JMESPathCheck('length(eventHubReceivers)', 0),
-            JMESPathCheck('location', 'swedencentral'),
+            JMESPathCheck('location', 'SwedenCentral'),
             #JMESPathCheck('name', action_group_name[:10]),
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_action_groups.py:137]: There're several comments need to be cleaned up in this PR.

**Review by necusjz — APPROVED** [review]

**Pull request merged**